src/main/java/bio/terra/service/dataset/flight/ingest/IngestValidateAzureRefsStep.java

```diff
@@ -55,14 +55,15 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
     // Then probe the file system to validate that the file exists and is part
     // of this dataset. We check all ids and return one complete error.
 
-    Set<String> invalidRefIds =
+    Set<InvalidRefId> invalidRefIds =
         table.getColumns().stream()
             .map(Column::toSynapseColumn)
             .filter(Column::isFileOrDirRef)
             .flatMap(
                 column -> {
                   List<String> refIdArray = azureSynapsePdao.getRefIds(tableName, column);
```

Comment:
I think I asked this before, but why do we validate refIds on a per-column basis? If the same refId is used in different columns, doesn't that result in an unnecessary call to the `tableServiceClient`? I would expect this code to generate a set of all refIds first, and then validate them with a single DAO call.

Reply:
I made a ticket to make the check more efficient: https://broadworkbench.atlassian.net/browse/DR-2199